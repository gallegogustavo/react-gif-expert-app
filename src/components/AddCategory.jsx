import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');

  {/* en los parámetros siempre se recibe el event y el target, es el método desestrcuturado*/}
  const onInputChange = ({ target }) => {
    setInputValue( target.value );  
  }

  const onSubmit = ( event ) => {
    event.preventDefault(); {/*evita el reload de toada la página al presiona enter */}
    if ( inputValue.trim().length <= 1) return;

    //setCategories( categories => [ inputValue, ...categories ] );
    onNewCategory( inputValue.trim() );
    setInputValue('');
  }

  return (
    <form onSubmit={ (event) => onSubmit(event)}>
      <input 
        type="text"
        placeholder="Buscar gifs"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form> 
  )
}
