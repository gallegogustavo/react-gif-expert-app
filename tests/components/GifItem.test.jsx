import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => { 

  const title = 'Titulo';
  const url = 'https://www.wikipedia.com';

  test('debe de hacer match con el sanpsho', () => {
     
    const { container } = render( <GifItem title={ title } url={ url } /> );
    //screen.debug();
    expect ( container ).toMatchSnapshot();
    
  })

  test('Debe mostrar el tÃ­tulo en el componente', () => { 

      render( <GifItem title={ title } url={ url } /> );
      expect( screen.getByText( title )).toBeTruthy(); 

  })
   
  test('debe de mostrar la imagen con el url y el alt indicado', () => { 

    render( <GifItem title={ title } url={ url } /> );
    
    //expect ( screen.getAllByRole('img').src ).toBe( url );

    const { src, alt } = screen.getByRole('img');
    screen.debug();
    
    expect( alt ).toBe( alt );
    //expect( src ).toBe( url );

  })

})