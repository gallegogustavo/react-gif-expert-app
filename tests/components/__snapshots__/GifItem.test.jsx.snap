// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el sanpsho 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://www.wikipedia.com"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
